support/src/deposit.rs

```diff
@@ -48,9 +48,9 @@ pub fn free_deposit<Account, Currency: Mutate<Account>>(
 	deposit: &Deposit<Account, Currency::Balance>,
 	reason: &Currency::Reason,
 ) -> DispatchResult {
-	let result = Currency::release(reason, &deposit.owner, deposit.amount, Precision::Exact);
+	let result = Currency::release(reason, &deposit.owner, deposit.amount, Precision::BestEffort);
 	debug_assert!(
-		result.is_ok(),
+		result.is_ok() && result.unwrap() == deposit.amount,
```

Comment:
No `unwrap()` please. Use optional binding, like `Some(amount) = deposit.amount`.

Reply:
I was thinking it is okay in this case since I check first if result is okay, and then I unwrap it. Because of the lazy evaluation of Rust, it should never panic. But removed anyway in: [06dea88](https://github.com/KILTprotocol/kilt-node/pull/527/commits/06dea8899c33d1d39b8988deb6260df441dbd2df).

Reply:
You don't need to check for `is_ok()` anymore. Plus, if you still call `result.unwrap()` when the assert fails, the code will panic.

This should work better:

```
debug_assert!(
  result == Ok(deposit.amount),
  "Released deposit amount does not match with expected amount. Expected: {:?}, Released result: {:?}",
  Ok::<_, DispatchError>(deposit.amount),
  result
);
```